compiler/src/dotty/tools/dotc/transform/TailRec.scala

```diff
@@ -32,10 +34,29 @@ import reporting.diagnostic.messages.TailrecNotApplicable
  *             contain such calls are not transformed).
  *             </p>
  *             <p>
- *             Self-recursive calls in tail-position are replaced by jumps to a
- *             label at the beginning of the method. As the JVM provides no way to
- *             jump from a method to another one, non-recursive calls in
- *             tail-position are not optimized.
+ *             When a method contains at least one tail-recursive call, its rhs
+ *             is wrapped in the following structure:
+ *             </p>
+ *             <pre>
+ *             var localForParam1: T1 = param1
+ *             ...
+ *             while (true) {
+ *               tailResult[ResultType]: {
+ *                 return {
+ *                   // original rhs
+ *                 }
+ *               }
+ *             }
+ *             </pre>
+ *             <p>
```

Comment:
While you're at it, you could remove the html tags in this comment, it should use markdown instead.